addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1038 @@
+    color_partner_id = fields.Many2one(related='user_id.partner_id', string="Color index of creator")  # Color of creator
+    active = fields.Boolean(default=True, help="If the active field is set to false, it will allow you to hide the event alarm information without removing it.")
+    categ_ids = fields.Many2many('calendar.event.type', 'meeting_category_rel', 'event_id', 'type_id', 'Tags')
+    attendee_ids = fields.One2many('calendar.attendee', 'event_id', 'Attendees', ondelete='cascade')
+    partner_ids = fields.Many2many('res.partner', 'calendar_event_res_partner_rel', string='Attendees', default=_get_default_partners, states={'done': [('readonly', True)]})
+    alarm_ids = fields.Many2many('calendar.alarm', 'calendar_alarm_calendar_event_rel', string='Reminders', ondelete="restrict", copy=False)
+
+    def _compute_is_attendee(self):
+        for meeting in self:
+            meeting.is_attendee = bool(meeting._find_my_attendee())
+
+    def _compute_attendee_status(self):
+        for meeting in self:
+            attendee = meeting._find_my_attendee()
+            meeting.attendee_status = attendee.state if attendee else 'needsAction'
+
+    def _compute_display_time(self):
+        for meeting in self:
+            meeting.display_time = self._get_display_time(meeting.start, meeting.stop, meeting.duration, meeting.allday)
+
+    @api.depends('start_date', 'allday', 'start_datetime')
+    def _compute_start(self):
+        for meeting in self:
+            meeting.start = meeting.start_date if meeting.allday else meeting.start_datetime
+            meeting.display_start = meeting.start_date if meeting.allday else meeting.start_datetime
+
+    @api.depends('stop_date', 'allday', 'stop_datetime')
+    def _compute_stop(self):
+        for meeting in self:
+            meeting.stop = meeting.stop_date if meeting.allday else meeting.stop_datetime
+
+    @api.constrains('start_datetime', 'stop_datetime', 'start_date', 'stop_date')
+    def _check_closing_date(self):
+        for event in self:
+            if event.stop < event.start:
+                raise ValidationError(_('Error ! End date cannot be set before start date.'))
+
+    @api.onchange('allday')
+    def onchange_allday(self):
+        if not ((self.start_date and self.stop_date) or (self.start and self.stop)):  # At first intialize, we have not datetime
+            return
+
+        if self.allday:  # from datetime to date
+            start_datetime = self.start_datetime or self.start
+
+            if start_datetime:
+                start = fields.Datetime.from_string(start_datetime)
+                self.start_date = fields.Date.to_string(start)
+
+            stop_datetime = self.stop_datetime or self.stop
+            if stop_datetime:
+                stop = fields.Datetime.from_string(stop_datetime)
+                self.stop_date = fields.Date.to_string(stop)
```

Comment:
``` self.stop_date = stop_datetime ``` should do the job why you need to convert it to from_string and then to_string(why we need to format it to server datetime format and then date format) ?
(applied to start_date also)